src/main/java/org/spongepowered/api/world/extent/Extent.java

```diff
@@ -32,4 +34,22 @@
  */
 public interface Extent extends EntityUniverse, WeatherUniverse, BlockVolume, BiomeArea {
 
+    /**
+     * Get a representation of the block at the given position.
+     *
+     * @param position The position
+     * @return The block
+     */
+    BlockLoc getFullBlock(Vector3i position);
```

Comment:
I was originally going to do this, but that would mean that BlockVolume has getBlockState instead of getBlock, which I find odd since it's a BlockVolume, not a BlockStateVolume. I don't think BlockLoc should be considered a Block, but instead an extension of it. BlockState should be what getBlock returns.